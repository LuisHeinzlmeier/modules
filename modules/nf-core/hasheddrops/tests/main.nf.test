nextflow_process {

    name "Test Process HASHEDDROPS"
    script "../main.nf"
    process "HASHEDDROPS"

    tag "modules"
    tag "modules_nfcore"
    tag "hasheddrops"
    tag "untar"

// file(params.modules_testdata_base_path + '/genomics/homo_sapiens/10xgenomics/cellranger/hashing_demultiplexing/htodemux.rds', checkIfExists: true)
        test("hto_matrix - runEmptyDrops:false - rna_matrix") {

        setup {
            run("UNTAR") {
                script "modules/nf-core/untar/main.nf"
            input:
            Channel.of(
                [ [id: 'hto'], file(params.modules_testdata_base_path + "/genomics/homo_sapiens/10xgenomics/cellranger/hashing_demultiplexing/hto.tar.gz") ],
                [ [id: 'rna'], file(params.modules_testdata_base_path + "/genomics/homo_sapiens/10xgenomics/cellranger/hashing_demultiplexing/rna.tar.gz") ]
            )
            }
        }

        when {
            process {
                """
                input[0] = UNTAR.out.untar.filter{ meta, _f -> meta.id == 'hto' }
                                       .combine(UNTAR.out.untar.filter{ meta, _f -> meta.id == 'rna' }
                                       .map{ _meta_hto, hto, _meta_rna, rna -> [[id: 'test], hto, "FALSE", rna] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("hto_matrix - runEmptyDrops:true - rna_matrix") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    "/Users/luisheinzlmeier/Desktop/test_data/hto",
                    "FALSE",
                    "/Users/luisheinzlmeier/Desktop/test_data/rna",
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("hto_matrix - runEmptyDrops - rna_matrix - stub ") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    "/Users/luisheinzlmeier/Desktop/test_data/hto",
                    "TRUE",
                    "/Users/luisheinzlmeier/Desktop/test_data/rna",
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}

