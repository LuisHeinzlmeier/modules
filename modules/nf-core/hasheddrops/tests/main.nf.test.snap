{
    "sarscov2 - bam - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "empty_drops_csv": [
                    
                ],
                "empty_drops_plot": [
                    
                ],
                "empty_drops_rds": [
                    
                ],
                "params": [
                    
                ],
                "plot": [
                    
                ],
                "rds": [
                    
                ],
                "results": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-17T19:41:18.570776"
    },
    "hto_matrix - runEmptyDrops - rna_matrix - stub ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test_results_hasheddrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hasheddrops.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test"
                        },
                        "test_plot_hasheddrops.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test"
                        },
                        "test_params_hasheddrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    "versions.yml:md5,ce84fb45f93458d1aa609015a25e7777"
                ],
                "empty_drops_csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "empty_drops_plot": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "empty_drops_rds": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "params": [
                    [
                        {
                            "id": "test"
                        },
                        "test_params_hasheddrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "plot": [
                    [
                        {
                            "id": "test"
                        },
                        "test_plot_hasheddrops.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "rds": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hasheddrops.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test"
                        },
                        "test_results_hasheddrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ce84fb45f93458d1aa609015a25e7777"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-17T21:37:16.241221"
    },
    "hto_matrix - runEmptyDrops - rna_matrix": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test_results_hasheddrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hasheddrops.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test"
                        },
                        "test_plot_hasheddrops.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test"
                        },
                        "test_params_hasheddrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    "versions.yml:md5,ce84fb45f93458d1aa609015a25e7777"
                ],
                "empty_drops_csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "empty_drops_plot": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "empty_drops_rds": [
                    [
                        {
                            "id": "test"
                        },
                        "test_emptyDrops.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "params": [
                    [
                        {
                            "id": "test"
                        },
                        "test_params_hasheddrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "plot": [
                    [
                        {
                            "id": "test"
                        },
                        "test_plot_hasheddrops.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "rds": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hasheddrops.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test"
                        },
                        "test_results_hasheddrops.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ce84fb45f93458d1aa609015a25e7777"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-17T21:33:51.33349"
    },
    "hto_matrix - runEmptyDrops:true - rna_matrix": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "empty_drops_csv": [
                    
                ],
                "empty_drops_plot": [
                    
                ],
                "empty_drops_rds": [
                    
                ],
                "params": [
                    
                ],
                "plot": [
                    
                ],
                "rds": [
                    
                ],
                "results": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-17T21:37:55.295104"
    },
    "hto_matrix - runEmptyDrops - rna_matrix:false": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "empty_drops_csv": [
                    
                ],
                "empty_drops_plot": [
                    
                ],
                "empty_drops_rds": [
                    
                ],
                "params": [
                    
                ],
                "plot": [
                    
                ],
                "rds": [
                    
                ],
                "results": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-17T20:00:18.443822"
    },
    "sarscov2 - bam": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "empty_drops_csv": [
                    
                ],
                "empty_drops_plot": [
                    
                ],
                "empty_drops_rds": [
                    
                ],
                "params": [
                    
                ],
                "plot": [
                    
                ],
                "rds": [
                    
                ],
                "results": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-17T18:27:50.319489"
    },
    "hto_matrix - runEmptyDrops:true - rna_matrix:false": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "empty_drops_csv": [
                    
                ],
                "empty_drops_plot": [
                    
                ],
                "empty_drops_rds": [
                    
                ],
                "params": [
                    
                ],
                "plot": [
                    
                ],
                "rds": [
                    
                ],
                "results": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-17T20:01:13.205588"
    },
    "hto_matrix - runEmptyDrops:false - rna_matrix": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "empty_drops_csv": [
                    
                ],
                "empty_drops_plot": [
                    
                ],
                "empty_drops_rds": [
                    
                ],
                "params": [
                    
                ],
                "plot": [
                    
                ],
                "rds": [
                    
                ],
                "results": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-17T20:00:58.945394"
    }
}